synapse/handlers/sync.py

```diff
@@ -206,9 +204,41 @@ class _RoomChanges:
 
     room_entries: List["RoomSyncResultBuilder"]
     invited: List[InvitedSyncResult]
+    """Our outstanding invitations at the `now_token`."""
+
     knocked: List[KnockedSyncResult]
+    """Rooms we have knocked on at the `now_token`."""
+
     newly_joined_rooms: List[str]
+    """Rooms we joined at some point between `since` and `now`.
+
+    Note: we need not be joined to these rooms at the `since` or `now` tokens.
+    Some examples:
+
+    Since       Midway    Now
+    --------------------------
+    <none>                join
+    invite                join
+    join        leave     join
+    invite      join      leave
+    """
     newly_left_rooms: List[str]
```

Comment:
...for consistency with the surrounding docstrings

```suggestion
    """

    newly_left_rooms: List[str]
```